### PR TITLE
Improve Rectangle HitShape customizability

### PR DESCRIPTION
**Pull request by reaperrr**
As temporary solution until someone adds a more customizable PolygonShape, I've added support for 
- applying the shape to all `TargetablePositions` (effectively implementing footprint shapes)
- rotating the shape by 90 degree (necessary for isometric perspective)

As example for the rotation, I'm using this to give TS walls an isometric, rectangular hitshape.

If you need a testcase for the functionality of applying the shape to all `TargetablePositions`, you can find that here: https://github.com/reaperrr/OpenRA/tree/hitshape-test

### CODE REVIEW

**Comment by obrakmann** [comment]
Looks good to me, :+1: 

The combat debug overlay doesn't quite do what I expected it to, though:
screenshot from 2016-07-02 17 07 52

**Comment by reaperrr** [comment]
> The combat debug overlay doesn't quite do what I expected it to, though:

Uh, yeah, that's definitely wrong.

**Comment by obrakmann** [comment]
Works fine with buildings on https://github.com/reaperrr/OpenRA/tree/hitshape-test, so there isn't anything completely wrong with it, I guess.

File: OpenRA.Mods.Common/HitShapes/Rectangle.cs [hunk 1]
```diff
@@ -89,24 +98,43 @@ public WDist DistanceFromEdge(WVec v)
...
 			var actorPos = actor.CenterPosition;
+			var orientation = new WRot(actor.Orientation.Roll, actor.Orientation.Pitch,
+				new WAngle(actor.Orientation.Yaw.Angle + (RotateToIsometry ? 128 : 0)));
 
-			var vertsTop = combatOverlayVertsTop.Select(v => wr.ScreenPosition(actorPos + v.Rotate(actor.Orientation)));
-			var vertsBottom = combatOverlayVertsBottom.Select(v => wr.ScreenPosition(actorPos + v.Rotate(actor.Orientation)));
-			wcr.DrawPolygon(vertsTop.ToArray(), 1, Color.Yellow);
-			wcr.DrawPolygon(vertsBottom.ToArray(), 1, Color.Yellow);
+			var targetablePositions = actor.TraitsImplementing<ITargetablePositions>();
+			if (ApplyToAllTargetablePositions && targetablePositions.Any())
```
Comment (obrakmann) [inline OpenRA.Mods.Common/HitShapes/Rectangle.cs]: Oh, I see. You effectively only draw anything when `ApplyToAllTargetablePositions` is set, and you didn't set it for gates.

**Comment by reaperrr** [comment]
Updated (and rebased).

**Comment by Mailaender** [comment]
image

image

Works as promised :white_check_mark:

**Comment by Mailaender** [comment]
No changelog as this is a temporary workaround.

**Pull request merged**